### PR TITLE
Fix wso2/product-is#9534: Support application/json content type in the request to the OAuth 2.0 token endpoint

### PR DESCRIPTION
**Pull request by omindu**
Fix wso2/product-is#9534

WIth the fix, requests to token endpoint supports application/json content type in addition to application/x-www-form-urlencoded

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth.common/src/test/java/org/wso2/carbon/identity/oauth/common/OAuthCommonUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+/**
+ * Test class for OAuthCommonUtil.
+ */
+public class OAuthCommonUtilTest {
+
+    @DataProvider(name = "Content Type Provider")
+    public Object[][] getContentType() {
+
+        return new Object[][]{
+                {"application/x-www-form-urlencoded", true},
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.common/src/test/java/org/wso2/carbon/identity/oauth/common/OAuthCommonUtilTest.java:41]: Can we add a test case for `application/json; charset=utf-8`

Comment (omindu) [reply components/org.wso2.carbon.identity.oauth.common/src/test/java/org/wso2/carbon/identity/oauth/common/OAuthCommonUtilTest.java:41]: Added test cases for `application/json; charset=utf-8`

**Comment by omindu** [comment]
PR builder passed - https://jenkins-support-all.wso2.com/jenkins/job/DEV_PR_BUILDER/852/

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/OAuthRequestWrapper.java [hunk 2]
```diff
@@ -32,15 +36,20 @@
  */
 public class OAuthRequestWrapper extends HttpServletRequestWrapper {
 
-    private MultivaluedMap<String, String> form;
+    private Map<String, List<String>> form;
     private Enumeration<String> parameterNames;
 
     public OAuthRequestWrapper(HttpServletRequest request, MultivaluedMap<String, String> form) {
```
[the remark this answers is not in the document]
Comment (omindu) [reply components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/OAuthRequestWrapper.java:43]: Done

**Review by pulasthi7 — APPROVED** [review]

**Review by mefarazath — APPROVED** [review]

**Pull request merged**